tests/javax/microedition/location/TestLocation.java

```diff
@@ -0,0 +1,50 @@
+package javax.microedition.location;
+
+import gnu.testlet.TestHarness;
+import gnu.testlet.Testlet;
+import gnu.testlet.TestUtils;
+import com.nokia.mid.location.LocationUtil;
+
+public class TestLocation implements Testlet, LocationListener {
+    TestHarness th;
+
+    public void locationUpdated(LocationProvider provider, final Location location) {
+        try {
+            QualifiedCoordinates c = location.getQualifiedCoordinates();
+            if (c != null) {
+                System.out.println("new Location Latitude: " + c.getLatitude() +
+                                   " Logitude: " + c.getLongitude());
+
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            th.fail("Unexpected exception: " + e);
+        }
+        // TODO There is an unknown bug that causes the following line blocking
+        // the VM.
+        // provider.setLocationListener(null, -1, -1, -1);
+        synchronized(this) {
+            this.notify();
+        }
+    }
+
+    public void providerStateChanged(LocationProvider provider, final int newState) {
+    }
+
+    public void test(TestHarness th) {
+        this.th = th;
+        try {
+            int[] methods = {(Location.MTA_ASSISTED | Location.MTE_SATELLITE | Location.MTY_TERMINALBASED)};
+            LocationProvider provider = LocationUtil.getLocationProvider(methods, null);
+            provider.setLocationListener(this, -1, -1, -1);
+            synchronized(this) {
+                this.wait();
+            }
+            th.pass();
```

Comment:
Could we use tests/index.js to override the real provider with a fake one for tests? Then we can test that this API works as expected, even if the data being provided is fake.